packages/salesforcedx-vscode-soql/test/vscode-integration/sfdx.test.ts

```diff
@@ -0,0 +1,27 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import * as sfdx from '../../src/sfdx';
+
+describe('sfdx utils should', () => {
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('displays an error if can not get connection to org', async () => {
+    sandbox.stub(sfdx.workspaceContext, 'getConnection').throws();
+    const vscodeErrorMessageSpy = sandbox.spy(
+      vscode.window,
+      'showErrorMessage'
+    );
+    await sfdx.withSFConnection(async () => {});
+    sfdx.debouncedShowChannelAndErrorMessage.flush();
```

Comment:
i found out that the debounce module adds a` flush() `method thankfully!

Reply:
i haven't tried this, but `@sinonjs/fake-timers` https://sinonjs.org/releases/v1.17.6/fake-timers/ seems to be the idiomatic way of doing this test

Reply:
@xyc - i tried that one but couldn't get it to work.  I tried with the whole fake clock and .tick(1000) and that api.  I have used that in the past.

Reply:
you might also try `import { debounce } from "debounce";` (so that it's not default export), and mock `debounce` to be `f` => f.call()`. Then we don't have to export `debouncedShowChannelAndErrorMessage` just for testing

Reply:
@xyc are you thinking something like [this](https://github.com/forcedotcom/soql-tooling/blob/198fe5596d3414f744ba82ec0168e59bfaa953b1/packages/soql-builder-ui/src/modules/querybuilder/whereModifierGroup/whereModifierGroup.test.ts#L15)?

Reply:
yeah, in sinon it's easier. Take a look at [fs stubs](https://github.com/forcedotcom/salesforcedx-vscode/blob/d1913dffa24eadc0e7d611ae675fa5b7704650e4/packages/salesforcedx-vscode-lwc/test/vscode-integration/testSupport/testIndexer/lwcTestIndexer.test.ts#L75-L78)

Reply:
Man, i battled this for a couple hours, having previously had success with sinon and setTimeouts.  I'll take another whack at it.

Reply:
So i revisited this again.  But sorry, I've spent time yesterday and today trying to use sinon FakeTimers and/or overwrite the prototype debounce to get this to work.  I just can't get that test to pass without using the provided flush() function of the debounce library.

Once difference between fs stubs and this module is it only exports a single default function, not an object.  I can't seem to intercept and/or overwrite that before it's cached by node.

Reply:
Sounds good, I'm fine with this too. I was thinking that debounce was using named exports `import { debounce } from 'debounce'` as well as default exports which might be utilized. But let's not spend more time on this